anyway/infographics_utils.py

```diff
@@ -556,61 +656,102 @@ def stats_accidents_by_car_type_with_national_data(
     try:
         number_of_years_ago = int(number_of_years_ago)
     except ValueError:
-        return {}
+        return None
     if number_of_years_ago < 0 or number_of_years_ago > 100:
-        return {}
+        return None
     location_info = extract_news_flash_location(news_flash_id)
     if location_info is None:
-        return {}
+        return None
     logging.debug("location_info:{}".format(location_info))
     location_text = get_news_flash_location_text(news_flash_id)
     logging.debug("location_text:{}".format(location_text))
     gps = location_info["gps"]
     location_info = location_info["data"]
-    output["meta"] = {"location_info": location_info.copy(), "location_text": location_text}
-    output["widgets"] = []
     resolution = location_info.pop("resolution")
     if resolution is None:
-        return {}
+        return None
 
     if all(value is None for value in location_info.values()):
-        return {}
+        return None
 
     last_accident_date = get_latest_accident_date(table_obj=AccidentMarkerView, filters=None)
     # converting to datetime object to get the date
     end_time = last_accident_date.to_pydatetime().date()
 
     start_time = datetime.date(end_time.year + 1 - number_of_years_ago, 1, 1)
 
-    # accident_severity count
-    items = get_accident_count_by_severity(
-        location_info=location_info,
+    request_params = RequestParams(
+        news_flash_id=news_flash_id,
         location_text=location_text,
+        location_info=location_info,
+        resolution=resolution,
+        gps=gps,
         start_time=start_time,
-        end_time=end_time,
-    )
+        end_time=end_time)
+    return request_params
+
+
+def create_infographics_data(news_flash_id, number_of_years_ago):
+    request_params = get_request_params(news_flash_id, number_of_years_ago)
+    location_info = request_params.location_info
+    resolution = request_params.resolution
+    start_time = request_params.start_time
+    end_time = request_params.end_time
+    location_text = request_params.location_text
+
+    if request_params is None:
+        return {}
 
-    accident_count_by_severity = Widget(name="accident_count_by_severity", rank=1, items=items)
-    output["widgets"].append(accident_count_by_severity.serialize())
+    output = {}
+    output["meta"] = {"location_info": location_info.copy(), "location_text": location_text}
+    output["widgets"] = []
+    output["widgets"].append(generate_widgets(request_params, get_widgets_def()))
+    # accident_severity count
+    # items = get_accident_count_by_severity(
+    #     location_info=location_info,
+    #     location_text=location_text,
+    #     start_time=start_time,
+    #     end_time=end_time,
+    # )
+    #
+    # accident_count_by_severity = Widget(name="accident_count_by_severity", rank=1, items=items)
+    # output["widgets"].append(accident_count_by_severity.serialize())
```

Comment:
is there a reason to keep all those comments?

Reply:
This is WIP. I want to have easy reference to how it was. Will remove eventually.

Reply:
ok, sorry